packages/zoe/src/contracts/loan/updateDebt.js

```diff
@@ -26,21 +28,22 @@ export const calculateInterest = (oldDebtValue, interestRate) =>
   );
 
 /** @type {MakeDebtCalculator} */
-export const makeDebtCalculator = debtCalculatorConfig => {
+export const makeDebtCalculator = async debtCalculatorConfig => {
   const {
     calcInterestFn = calculateInterest,
     originalDebt,
     loanMath,
     periodNotifier,
     interestRate,
     interestPeriod,
+    basetime,
     zcf,
     configMinusGetDebt,
   } = debtCalculatorConfig;
   let debt = originalDebt;
 
   // the last period-end for which interest has been added
-  let lastCalculationTimestamp;
+  let lastCalculationTimestamp = basetime;
```

Comment:
You said there was a bug where we calculated from the start of time, but in initialization in what was line 96 `lastCalculationTimestamp` is set to `value`, so I don't think that's true. I like this version better, though.